### PR TITLE
fix event modal popping up when clicking bookmark

### PR DESCRIPTION
**Pull request by margulies**
Fixes event modal popup bug when clicking on bookmark.

### CODE REVIEW

File: src/hooks/useShowHide.ts [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
-  const show = useCallback(() => {
-    setShown(true);
-  }, []);
+  const show = useCallback(
+    (e?) => {
+      if (targetClosest && e?.target.closest(targetClosest)) return;
+      setShown(true);
+    },
+    [targetClosest]
+  );
```
Comment (sunny-viktoryia) [inline src/hooks/useShowHide.ts:20]: It looks a bit hacky to my eyes. 
I see your implemented fix this way: we have some `targetClosest` exception. When the event was fired within that `targetClosest` selector we ignore the modal opening. Am I correct?

Let's consider another case. Let's say we want to make clickable not only the bookmark icon but the host in the event box in the schedule too.
image
It means we have 2 clickable child elements. And we don't want the event modal to open when we click on them. And in fact, we can have more complex containers with more than 2 clickable child elements. Your implement version of the hook can't handle that case.

You could suggest transforming `targetClosest` to the array of selectors. But I wouldn't support that. I suppose would be better to keep things as isolated as possible. The name of the hook implies a straightforward logic of having a boolean status: "is shown" / "is hidden". Adding feature-specific things would one day overcomplicate the hook and increase the possibility of a bug.

Although I like your idea of building a generic solution, currently I can't quickly come up with a better solution than explicitly add the exceptions to the functions where it's needed. 

@0xdevalias @mike-lvov would love to hear your thoughts. 🙃

**Review by sunny-viktoryia — COMMENTED** [review]
Posting here a comment to discuss the idea of the fix.

Thanks for your contributions, @margulies!

File: src/hooks/useShowHide.ts (hunk 1 above)
Comment (margulies) [reply src/hooks/useShowHide.ts:20]: All sounds totally reasonable. Just wanted to mention that the current implemtation should be able to handle an array of clickable child elements. For example, here: https://github.com/sparkletown/sparkle/blob/4e9c93cfe4b8d5f6e31a26fac8cb0c0f6fe8b040/src/components/templates/PosterHall/components/PosterPreview/PosterPreview.tsx#L50

By passing a string array to `targetClosest` similar to the example linked above, it should achieve a similar functionality. 

More broadly though, you're absolutely right that a more elegent general solution would be possible here.

Comment (margulies) [reply src/hooks/useShowHide.ts:20]: Oh! Sorry, one other question: the real goal of this is to deal with the bug in the event modal. If there's a simple way to add in the specific exception in `ScheduleEvent.tsx` without needing to copy over all the code from `useShowHide`, that's seems totally sufficient for the time being. Happy to do that if you could point me in the simplest direction to achieve that goal.

Comment (sunny-viktoryia) [reply src/hooks/useShowHide.ts:20]: I'd do something like this in the `ScheduleEvent.tsx`: 
```ts
  const onEventBoxClick = useCallback(
    (e) => {
      if (e.target.closest(".ScheduleEvent__bookmark")) return;

      showEventModal();
    },
    [showEventModal]
  );
  
  // some logic 
  
      <div className={containerClasses} onClick={onEventBoxClick}>
  // ...
```

Comment (margulies) [reply src/hooks/useShowHide.ts]: That was so much easier! Thanks! Cleaned up and much simpler now... really appreciate it the help

File: src/components/molecules/ScheduleEvent/ScheduleEvent.tsx [hunk 2]
```diff
@@ -63,6 +63,8 @@ export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
     containerCssVars
   );
 
+  const ScheduleEventBookmarkClass = "ScheduleEvent__bookmark";
```
Comment (sunny-viktoryia) [inline src/components/molecules/ScheduleEvent/ScheduleEvent.tsx:66]: can we move this static const out of the `ScheduleEvent` component not to create it on every render?
(would be better to see it right after the imports before the `ScheduleEventProps` interface)

**Review by sunny-viktoryia — APPROVED** [review]
Looks good to me (would be perfect to address my comment). 🙃

Thanks for providing the fix, @margulies.

File: src/components/molecules/ScheduleEvent/ScheduleEvent.tsx (hunk 2 above)
Comment (margulies) [reply src/components/molecules/ScheduleEvent/ScheduleEvent.tsx]: Done! 34b0660cab61d257db73fc7da72b9313eaa75589
Thanks! (and good to know how to creating a const on every render. It seems obvious now that you mention it, but hadn't realized it before.)

**Pull request merged**